cheatsheets_draft/Webhook_Security_Guidelines_CheatSheet.md

```diff
@@ -0,0 +1,169 @@
+
+Retry Logic - Have provision to reset a webhook's authentication key. To ensure that you do not lose any webhook requests between the time you reset your key and when you update your application to start using that new key, your webhook processor should reject requests with failed signatures with custom Status code. 
+
+Abuse Case: 
+
+- Using a targeted webhook, capture HTTP Request with Auth key. Generate new Authkey for the Webhook URL, try POST ing with expired Auth key, Invalid Authkey, payload and validate.
+
+
+## Control 10 - Pass Secrets in Request Header 
+
+Protection from repudiation attacks - Most webhook providers allow you to pass a secret or a "token" in every webhook request. Example Github. The secret is typically passed as an HTTP header, as a field in the JSON payload or appended to the request URL. When you confirm the token you received is the one you expected, it helps validate the request was sent by the webhook publisher. 
+
+Abuse Case: 
+
+- Using a targeted webhook capture a known good message. Tamper or remove the Token from Request Header and POST the message to target URL. The webhook should respond that the message was invalid. If it does not it is vulnerable to repudiation attacks
+
+
+## Control 11 - Fail Safe and Secure 
+
+Event Handling - Handling webhook events correctly is crucial to making sure your integration's business logic works as expected. Webhook endpoints might occasionally receive the same event more than once. We advise you to guard against duplicated event receipts by making your event processing idempotent. One way of doing this is logging the events you’ve processed, and then not processing already-logged events. 
+
+Abuse Case: 
+
+- Using a targeted webhook try to send incorrect, corrupted or flawed messages to the webhook targeting processing flaws in the webhook service or any services processing webhook messages. If the service does not handle the corrupt/flawed message it is vulnerable to this attack.
+
+
+## Control 12 - Replay protections 
+
+Replay Attacks - Anyone who is sniffing traffic can intercept the request and replay it. To protect against replay attacks, you can rely on adding timestamp for when the webhook event was generated. These should be added to the header. Timestamps are typically used as a seed for computing and verifying a signature. If you try to just change the timestamp, the signature verification will fail because it's not valid. 
+
+Abuse Case: 
+
+- Using a targeted webhook capture a webhook message and POST and replay the same message without modification. If you can do this, replay attacks are possible
+
+
+## Control 13 - Payload best practices 
+
+Service Attack Vector - In most cases webhooks are being used as input for a service account. In these scenarios you need to be mindful to validate the data coming using tight validations of the payload for injection, DDOS, XSS, and other attacks that can be found in the OWASP TOP 10. 
```

Comment:
OWASP Top 10 is not a checklist for best practices! WSTG is the web security testing guide which should be followed to test.
As we're in the cheat sheets project, we should be linking to the relevant cheat sheets. I can help you with that once we're done with reviewing the sheet's content.

Reply:
Thank you so much @ThunderSon . I have got this reviewed by few of my friends and colleagues who develop Webhooks and Abuse them mostly working in Appsec. Since this is for community all feedback is welcome and hope we can have go to cheat sheet for Webhooks :)

Reply:
Thank you for tackling this with your colleagues! Top 10 is a risk study, more so than a proper attack reference, that's what I simply meant 😄 [WSTG](https://github.com/OWASP/wstg) on the other hand is a full web security testing framework.